### ISSUE TITLE
querystring change

### ISSUE THREAD

**Issue by Ti-webdev** [sveltejs/sapper#558]
Hi
I have simple route:
```html
<a href="?category=filament&weight=300g">filament 300g</a>
<a href="?category=filament&weight=1kg">filament 1kg</a>
<pre>{ JSON.stringify(items, 0, 2) }</pre>

<script context="module">
export async function preload ({ query }) {
  console.log('preload')
  const res = await this.fetch(`/items.json?${new URLSearchParams(query)}`)

  if (res.status === 200) {
    return { items: await res.json() }
  } else {
    this.error(res.status)
  }
}
</script>

<script>
	export let items
</script>
```
For first open everything is fine.
When I click to any of link preload is not called but querystring in URL is changed.
How I should do that?

**Comment by SkaterDad** [comment]
I'm seeing this as well.  Here's a Codesandbox repro (see the blog section): https://codesandbox.io/s/6y330krxmz

Stepping through the code, I think this line is the problem:
https://github.com/sveltejs/sapper/blob/82e637ea7c7295cc4f84a3c4cc33c00ce281efb0/runtime/src/app/app.ts#L250
```js
// This evaluates True when the query param was the only change.
if (!session_dirty && current_branch[i] && current_branch[i].segment === segment) return current_branch[i];
```

**Comment by antonheryanto** [comment]
hit similar bug. it happen after #538 merge

**Comment by SkaterDad** [comment]
Working on a  PR for this.

Should query param changes trigger preload on all nested segments, or only the last one?

My own use case only needs the query string for the final segment.

**Comment by thgh** [comment]
Sort of related: I wish a store could be preloaded. Deriving from $page would then allow to implement something similar to the preload behavior.

```js
  export async function preload() {
    await $items
  }
  const items = derive(stores.page, page =>
    fetch(`/items.json?${new URLSearchParams(page.query)}`).then(r =>
      res.status === 200 ? r.json() : []
    )
  )
```

Or fancier:
```js
preload: items = derive(stores.page, page =>
  fetch(`/items.json?${new URLSearchParams(page.query)}`).then(r =>
    res.status === 200 ? r.json() : []
  )
)
```

**Comment by Conduitry** [comment]
Huh #701 was apparently a duplicate of this and no one noticed. It was fixed in #704 and released in 0.27.1.
